packages/astro/src/core/ssr/index.ts

```diff
@@ -266,7 +266,8 @@ export async function render(renderers: Renderer[], mod: ComponentInstance, ssrO
 
   // run transformIndexHtml() in dev to run Vite dev transformations
   if (mode === 'development') {
-    html = await viteServer.transformIndexHtml(viteifyURL(filePath), html, pathname);
+    const relativeURL = filePath.href.replace(astroConfig.projectRoot.href, '/');
+    html = await viteServer.transformIndexHtml(relativeURL, html, pathname);
```

Comment:
Big change 1: use project-relative URLs to generate the script proxies.

Note that `relativeURL` will be `/index.astro`, while `pathname` will be `/`. Both are needed because Vite doesn’t know `'.astro' === '.html'`.

Reply:
Note the `/:/` is because `C:/` isn’t the start of a browser-compatible pathname